packages/react-graphql-universal-provider/src/test/GraphQLUniversalProvider.test.tsx

```diff
@@ -64,22 +72,36 @@ describe('<GraphQLUniversalProvider />', () => {
     // client would typically read serializations from the DOM on initialization).
     mount(
       <HtmlContext.Provider value={htmlManager}>
-        <GraphQLUniversalProvider createClient={() => client} />
+        {graphQLProvider}
       </HtmlContext.Provider>,
     );
 
     expect(restoreSpy).toHaveBeenCalledWith(initialData);
   });
 
   it('renders an <Effect/> from ApolloBridge', () => {
-    const client = new ApolloClient({
+    const clientOptions = {
       cache: new InMemoryCache(),
       link: new ApolloLink(),
-    });
+    };
     const graphQL = mount(
-      <GraphQLUniversalProvider createClient={() => client} />,
+      <GraphQLUniversalProvider createClientOptions={() => clientOptions} />,
     );
 
     expect(graphQL).toContainReactComponent(Effect);
   });
+
+  it('includes createSsrExtractableLink if a link is not present', () => {
+    const clientOptions = {
+      cache: new InMemoryCache(),
+    };
+
+    const client = mount(
+      <GraphQLUniversalProvider createClientOptions={() => clientOptions} />,
+    )
+      .find(ApolloProvider)!
+      .prop('client');
+
+    expect(client.link instanceof SsrExtractableLink).toBe(true);
```

Comment:
You can do nicer assertions than this: expect(client).toContainReactComponent(ApolloProvider, {client: expect.any(SarExtractableLink)})

Reply:
That doesn't work since `client` expects an `ApolloClient` constructor.  Is there a way to perform nested assertions. Something like this (which doesn't work either) 

```
expect(client).toContainReactComponent(ApolloProvider, {
  client: {
      link: expect.any(SarExtractableLink),
  }
)
```

Reply:
sorry it would be `expect.objectContaining({link: expect.any(SsrExtractableLink)})`

Reply:
Going to merge, for now. Will follow up if I get it the cleaner assertion working